criu/pie/restorer.c

```diff
@@ -1327,17 +1327,24 @@ static int fd_poll(int inotify_fd)
  */
 #define EVENT_BUFF_SIZE ((sizeof(struct inotify_event) + PATH_MAX))
 
+/* We can get interrupted in poll, so try some more */
+#define MAX_POLL_RETRY 5
+
 /*
  * Read all available events from inotify queue
  */
 static int cleanup_inotify_events(int inotify_fd)
 {
 	char buf[EVENT_BUFF_SIZE * 8];
-	int ret;
+	int ret, retry = 0;
 
-	while (1) {
+	while (retry < MAX_POLL_RETRY) {
 		ret = fd_poll(inotify_fd);
```

Comment:
we call poll with zero timeout, it shouldn't block. why doesn't this mean that it should not be interrupted?

Reply:
I read the kernel code quickly and if I understand it right, EINTR can be returned only if there is no event to report by.